### PR TITLE
Bonded target

### PR DESCRIPTION
**Pull request by devintegral3**
- SFC is ownable. updateGasPowerAllocationRate and updateBaseRewardPerSec are callable by owner.
- rewards are claimable only after 6 months passed since enough coins are staked
- staker may have optional metadata

### CODE REVIEW

File: contracts/sfc/Staker.sol [hunk 1]
```diff
@@ -369,13 +406,13 @@ contract Stakers is StakersConstants {
 
         require(delegations[delegator].amount != 0, "delegation doesn't exist");
         require(delegations[delegator].deactivatedTime == 0, "delegation is deactivated");
+        require(checkBonded(), "not passed enough time since stake is bonded");
```
Comment (andrecronje) [inline contracts/sfc/Staker.sol]: Won't this immediately be true?

function checkBonded() internal returns (bool) {
	maintainCapReachedDate(); // capReachedDate would be 0
	return rewardsAllowed(); //block.timestamp.sub(0) >= 6 months == true
}

function maintainCapReachedDate() internal {
	uint256 current = bondedRatio();
	uint256 target = bondedTargetRewardUnlock();
	if (current >= target && capReachedDate == 0) {
		changeCapReachedDate(block.timestamp);
	} else if (current < target && capReachedDate != 0) {
		changeCapReachedDate(0);
	}
}
function rewardsAllowed() public view returns (bool) {
	return block.timestamp.sub(capReachedDate) >= unbondingUnlockPeriod();
}

function unbondingUnlockPeriod() public pure returns (uint256) {
  return 60 * 60 * 24 * 30 * 6; // 6 months
}

File: contracts/sfc/Staker.sol [hunk 2]
```diff
@@ -369,13 +406,13 @@ contract Stakers is StakersConstants {
...
 
         uint256 pendingRewards;
         uint256 fromEpoch;
         uint256 untilEpoch;
         (pendingRewards, fromEpoch, untilEpoch) = calcDelegationRewards(delegator, _fromEpoch, maxEpochs);
 
-        require(bondedRatio > bondedTargetRewardUnlock(), "below minimum bonded ratio");
         require(block.timestamp > capReachedDate + unbondingUnlockPeriod(), "before minimum unlock period");
```
Comment (andrecronje) [inline contracts/sfc/Staker.sol]: require(block.timestamp > capReachedDate + unbondingUnlockPeriod(), "before minimum unlock period");

block.timestamp > 0 + 6 months == true

File: contracts/sfc/Staker.sol (hunk 1 above)
Comment (devintegral3) [reply contracts/sfc/Staker.sol]: Should be fixed in ded1f86, plesae check

File: contracts/sfc/Staker.sol (hunk 2 above)
Comment (devintegral3) [reply contracts/sfc/Staker.sol]: Should be fixed in ded1f86, please check

**Comment by devintegral3** [comment]
There were issue that capReachedDate could be updated only during reward claiming, but during reward claiming cap was required to be 6 months in past. So added additional method to maintain capReachedDate in #1dc507

**Pull request merged**